### DISCUSSION TITLE
Docs: mention TypeScript language support, change sidebar

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#10735] [Suggest an Idea]
We do support the TypeScript language, via the `npm` or `yarn` package managers, but we don't explicitly list it in the list of supported languages. Maybe we want to list it?

We could make the sidebar look like this:

Language support:
  - Items
  - JavaScript/Typescript
  - Items

And expand the text in the JavaScript section to say something like:

```markdown

## JavaScript/TypeScript

Renovate supports upgrading JavaScript dependencies specified in package.json files.
Renovate also supports the TypeScript language, as you install the language via `npm` or `yarn`.

npm, yarn, and pnpm are all supported.
```

Link to doc on JavaScript language support: https://docs.renovatebot.com/javascript/

**Comment by viceice** [comment]
🤔 I don't see any benefit, because we support the javascript based managers (npm / yarn / pnpm). we don't support typescript as a direct language. we also don't directly support updateing javascript directly.

So i'm not sure if this helps. What do you think we support via installing typescript dependency?

**Reply by HonkingGoose** [reply]
We call the section _Language support_, which to me, implies: "Programming languages you can use with Renovate bot", and it does kinda-sorta-work when read that way. But TypeScript is a programming language you can use with Renovate (install `tsc` via `npm` say), but we don't mention that language explicitly.

What we actually seem to mean is more like _package managers we support for each programming language_, but that's not what the sidebar heading says. 😉 

For JavaScript, we don't "support" the language, but only the JavaScript package managers `npm`/`yarn`/`pnpm`.
For Docker we actually do seem to support things like Helm/Kubernetes/Ansible.

I think we not properly surfacing what we actually support by using the headings in this way?

---

How about making the sidebar like this?

**Package manager support**

- Npm
- Yarn
- Docker
- Kubernetes
- Helm
- Gradle
- Maven

So each manager gets it own section.
That way you can expand the sidebar, and get a direct view of all things we support.
So you don't have to go rooting around in the Docker document to find out we support Helm/Kubernetes.
This would require shuffling around the content to match the new sidebar/file structure.

**Reply by viceice** [reply]
🤔  We have some kind of manager list at https://docs.renovatebot.com/modules/manager/

Maybe we should move the language support stuff to the relevant manager and make the managers section expandable?

**Reply by HonkingGoose** [reply]
> 🤔 We have some kind of manager list at https://docs.renovatebot.com/modules/manager/

Oh yeah, that's the real summary of managers we support! Forgot about that one... 😄 

> Maybe we should move the language support stuff to the relevant manager and make the managers section expandable?

The full list of managers is way too long to show on the sidebar by default, so the section should expand if needed.
Having a list of "languages" is still handy, in case you know "I'm using Python" but don't know if you use `pip` or something else.
So I'm not sure moving it is the full answer to the problem...

---

It's a tricky problem, I think the root cause of my confusion is that it's really tricky to get a good overview of the package manager/language support just by browsing/skimming the docs.

Maybe collapsing the "Modules -> Managers" _and_ the "Language support" sections into a single big "Stuff we support" page would be better? But that risks making a page that's too big to navigate.... 😄 

---

@rarkins What do you think of our current categorization?
